src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfToken.java

```diff
@@ -0,0 +1,152 @@
+import org.opensearch.threadpool.ThreadPool;
+
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class CreateOnBehalfOfToken extends BaseRestHandler {
+
+    private JwtVendor vendor;
+    private final ThreadPool threadPool;
+
+    private ConfigModel configModel;
+
+    private DynamicConfigModel dcm;
+
+    @Subscribe
+    public void onConfigModelChanged(ConfigModel configModel) {
+        this.configModel = configModel;
+    }
+
+    @Subscribe
+    public void onDynamicConfigModelChanged(DynamicConfigModel dcm) {
+        this.dcm = dcm;
+        if (dcm.getDynamicOnBehalfOfSettings().get("signing_key") != null && dcm.getDynamicOnBehalfOfSettings().get("encryption_key") != null) {
+            this.vendor = new JwtVendor(dcm.getDynamicOnBehalfOfSettings(), Optional.empty());
+        } else {
+            this.vendor = null;
+        }
+    }
+
+    public CreateOnBehalfOfToken(final Settings settings, final ThreadPool threadPool) {
+        this.threadPool = threadPool;
+    }
+
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    @Override
+    public List<Route> routes() {
+        return addRoutesPrefix(
+                ImmutableList.of(
+                        new Route(Method.POST, "/user/onbehalfof")
+                )
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        switch (request.method()) {
+            case POST:
+                return handlePost(request, client);
+            default:
+                throw new IllegalArgumentException(request.method() + " not supported");
+        }
+    }
+
+    private RestChannelConsumer handlePost(RestRequest request, NodeClient client) throws IOException {
+        return new RestChannelConsumer() {
+            @Override
+            public void accept(RestChannel channel) throws Exception {
+                final XContentBuilder builder = channel.newBuilder();
+                BytesRestResponse response;
+                try {
+                    if (vendor == null) {
+                        channel.sendResponse(new BytesRestResponse(RestStatus.SERVICE_UNAVAILABLE, "on_behalf_of configuration is not being configured"));
+                        return;
+                    }
+
+                    final Map<String, Object> requestBody = request.contentOrSourceParamParser().map();
+                    final String reason = (String)requestBody.getOrDefault("reason", null);
+
+                    final Integer tokenDuration = Optional.ofNullable(requestBody.get("duration"))
+                            .map(value -> (String)value)
+                            .map(Integer::parseInt)
+                            .map(value -> Math.min(value, 72 * 3600)) // Max duration is 72 hours
+                            .orElse(24 * 3600); // Fallback to default;
```

Comment:
Seems quite long--I was under the impression we were thinking 5 minutes or so? Could this not lead to up to 3 days of an invalid role being active?

Reply:
Ya this is the one I will modify it to something like 5 mins (300 seconds). I will do that in the next couple commits.